library/src/main/scala/scala/scalajs/js/Generator.scala

```diff
@@ -0,0 +1,60 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js API               **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2015, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+package scala.scalajs.js
+
+import scala.language.implicitConversions
+
+import scala.annotation.implicitNotFound
+import scala.annotation.unchecked.uncheckedVariance
+
+import scala.scalajs.js
+import js.annotation._
+
+/** <span class="badge badge-ecma6" style="float: right;">ECMAScript 6</span>
+ *  JavaScript Generator, such as the result of generator function.
+ *
+ *  @tparam V The type of elements returned by `next()`
+ *  @tparam I The type of values expected by `next()` as parameter
+ *  @tparam R The type of the result/done value.
```

Comment:
I do not understand what you gain by making the distinction between `V` and `R` here...